### DISCUSSION TITLE
执行用户脚本是如何实现的？

### DISCUSSION THREAD

**Discussion by suzikuo** [scriptscat/scriptcat#332] [General]
执行用户脚本是如何实现的？
由于csp的限制，无法在content.js中创建script标签，以及执行eval/new Function等。那么是如何执行用户脚本的呢？

小白求指教~~~

**Comment by CodFrm** [comment]
mv2的时候比较暴力，直接使用`chrome.webrequest`将csp策略给移除了，mv3可以通过`chrome.userScript`绕过这个策略注册脚本（不过我还没测试这种情况，哥哥可以自行试一下先）

**Reply by suzikuo** [reply]
好的好的，谢谢大佬
